app/services/container_topology_service.rb

```diff
@@ -117,35 +117,30 @@ def build_entity_data(entity)
     data
   end
 
-  def entity_status(entity, kind)
-    case kind
-    when 'Vm', 'Host' then entity.power_state.capitalize
-    when 'ContainerNode'
-      ready_status = 'Unknown'
+  def entity_status(entity)
+    if entity.kind_of?(Host) || entity.kind_of?(Vm)
+      status = entity.power_state.capitalize
+    elsif entity.kind_of?(ContainerNode)
+      status = 'Unknown'
       entity.container_conditions.each do |condition|
         if condition.try(:name) == 'Ready' && condition.try(:status) == 'True'
-          ready_status = 'Ready'
+          status = condition.name
         else
-          ready_status = 'NotReady'
+          status = 'NotReady'
         end
       end
-      ready_status
-    when 'ContainerGroup' then entity.phase
-    when 'Container' then entity.state.capitalize
-    when 'ContainerReplicator'
-      if entity.current_replicas == entity.replicas
-        'OK'
-      else
-        'Warning'
-      end
-    when 'ContainerManager'
-      if entity.authentications.empty?
-        'Unknown'
-      else
-        entity.authentications.first.status.capitalize
-      end
-    else 'Unknown'
+    elsif entity.kind_of?(ContainerGroup)
+      status = entity.phase
+    elsif entity.kind_of?(Container)
+      status = entity.state.capitalize
+    elsif entity.kind_of?(ContainerReplicator)
+      status = (entity.current_replicas == entity.replicas) ? 'OK' : 'Warning'
+    elsif entity.kind_of?(ManageIQ::Providers::ContainerManager)
+      status = entity.authentications.empty? ? 'Unknown' : entity.authentications.first.status.capitalize
+    else
+      status = 'Unknown'
```

Comment:
It looks like we need a status/state mixin or something that adds a status method with "Unknown" by default and move each of the class's logic into a status method or something.  Again, we'd need to look at how the UI presents this information in the other places... If they're hardcoding `power_state`, they could also leverage a more general `status` method that calls `power_state` for the more legacy models like `Vm` or `Host`.

This PR is clearly better than it was though so :+1: